device/simulation/tt_simulation_host.hpp

```diff
@@ -20,6 +20,6 @@ class tt_SimulationHost {
     void send_to_device(uint8_t *buf, size_t buf_size);
     size_t recv_from_device(void **data_ptr);
 private:
-    nng_socket host_socket;
-    nng_dialer host_dialer;
+    nng_socket *host_socket;
+    nng_dialer *host_dialer;
```

Comment:
Can you use `std::unique_ptr<nng_socket>` to avoid having to worry about memory management?

Reply:
🫡